app/models/miq_schedule_worker/runner.rb

```diff
@@ -234,11 +234,22 @@ def schedules_for_scheduler_role
       end
     end
 
-    # run run chargeback generation every day at specific time
+    # run chargeback generation every day at specific time
     schedule_chargeback_report_for_service_daily
+
+    schedule_check_for_task_timeout
+
     @schedules[:scheduler]
   end
 
+  def schedule_check_for_task_timeout
+    every = worker_settings[:task_timeout_check_frequency]
+    scheduler = scheduler_for(:scheduler)
+    scheduler.schedule_every(every, :first_at => Time.current + 1.minute) do
```

Comment:
i've done manual testing: leaving active records in DB and restarting server - works